### PR TITLE
Split DigitizeButton

### LINKED ISSUE #475: Split Digitize Button
We should split the `DigitzeButton` into multiple `buttons`.

At least into `DrawButton` and `ModifyButton`, `CopyButton` and `DeleteButton`.

~~It also contains a prop named `style` which is allready reserverd for css inline-styles. It should be removed. See https://github.com/terrestris/react-geo/blob/master/src/Button/DigitizeButton/DigitizeButton.jsx#L289~~

### PR DESCRIPTION
**Pull request by simonseyock**
### Description:

This PR extracts the draw parts from DigitizeButton into a seperate component as described in #475.

Major differences to the DigitizeButton:

1. It accepts a `layer` option instead of `layerName` option.
2. The `drawStyle` is no longer applied to the layer. To change the style of the layer use:
  ```
  const digitizeLayer = DigitizeUtil.getDigitizeLayer(map);
  digitizeLayer.setStyle(...);
  ```
3. It will always use an own interaction and will not share it with other buttons of the same type. (It might happen that the `DigitizeButton` tries to use same interaction - so it should be avoided to use a `DigitizeButton` for drawing if you intend to use a `DrawButton`).

This PR includes useful helper functions to trigger events on the open layers map.

### CODE REVIEW

**Comment by simonseyock** [comment]
I was able to extract the DrawButton. And it works:

image

**Comment by simonseyock** [comment]
Major differences to the DigitizeButton:

1. It accepts a `layer` option instead of `layerName` option.
2. The `drawStyle` is no longer applied to the layer. To change the style of the layer use:
  ```
  const digitizeLayer = DigitizeUtil.getDigitizeLayer(map);
  digitizeLayer.setStyle(...);
  ```

This PR includes useful helper functions to trigger events on the open layers map.

File: src/Button/DrawButton/DrawButton.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+import MapUtil from '@terrestris/ol-util/dist/MapUtil/MapUtil';
+
+import OlView from 'ol/View';
+import OlMap from 'ol/Map';
+import MapComponent from '../../Map/MapComponent/MapComponent';
+import { clickMap, doubleClickMap } from '../../Util/rtlTestUtils';
+import LineString from 'ol/geom/LineString';
+import Polygon from 'ol/geom/Polygon';
+
+
```
Comment (LukasLohoff) [inline src/Button/DrawButton/DrawButton.spec.tsx:16]:
```suggestion
```

File: src/Button/DrawButton/DrawButton.tsx [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        setShowLabelPrompt(true);
+        evt.feature.set('isLabel', true);
+        setDigitizeTextFeature(evt.feature);
+      });
+    }
+
+    return () => {
+      unByKey(key);
+      map.removeInteraction(interaction);
+    }
```
Comment (LukasLohoff) [inline src/Button/DrawButton/DrawButton.tsx:200]:
```suggestion
    };
```

**Review by LukasLohoff — APPROVED** [review]
:medal_sports:

File: src/Button/DrawButton/DrawButton.tsx [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+  /**
+   * Currently drawn feature which should be represent as label or postit.
+   */
+  const [digitizeTextFeature, setDigitizeTextFeature] = useState<OlFeature<OlGeometry>>(null);
+
+  /**
+   * Called when the digitize button is toggled. If the button state is pressed,
+   * the appropriate draw, modify or select interaction will be created.
+   * Otherwise, by untoggling, the same previously created interaction will be
+   * removed from the map.
```
Comment (annarieger) [inline src/Button/DrawButton/DrawButton.tsx:224]: I think this comment is a bit misleading

File: src/Button/DrawButton/DrawButton.tsx [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+  return (
+    <span className={btnWrapperClass}>
+      <ToggleButton
+        onToggle={onToggleInternal}
+        className={finalClassName}
+        {...passThroughProps}
+      />
+      {
+        showLabelPrompt ?
```
Comment (annarieger) [inline src/Button/DrawButton/DrawButton.tsx:278]: absolutely minor:

can be simplified to `showLabelPrompt && <Modal [...] />`

File: src/Util/DigitizeUtil.ts [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+      }),
+      stroke: new OlStyleStroke({
+        color: DigitizeUtil.DEFAULT_STROKE_COLOR
+      })
+    })
+  });
+
+  static DIGITIZE_LAYER_NAME = 'react-geo_digitize';
+
+  static getDigitizeLayer(map: OlMap): OlVectorLayer<OlSourceVector<Geometry>> {
```
Comment (annarieger) [inline src/Util/DigitizeUtil.ts:86]: A tiny comment would be useful :)

**Review by annarieger — APPROVED** [review]
Very nice work! 🍍

**Pull request merged**